### PR TITLE
fix(auth): deprecate `FirebaseAuth.instanceFor`'s `persistence` parameter

### LINKED ISSUE #10961: 🐛 [firebase_auth] Flutter web: Persistence not working "Firebase: initializeAuth() has already been called with different options."
Reopening https://github.com/firebase/flutterfire/issues/10335 

In short, login doesn't work using **persistence** in web like `_auth = firebase_auth.FirebaseAuth.instanceFor(app: firebaseApp, persistence: firebase_auth.Persistence.INDEXED_DB);` 

It outputs the following when signing in

<details>
<summary>Error stack</summary>

CRASHLYTICS Recording Error: Auth error google FirebaseError: Firebase: initializeAuth() has already been called with different options. To avoid this error, call initializeAuth() with the same options as when it was originally called, or call getAuth() to return the already initialized instance. (auth/already-initialized).
CRASHLYTICS Trace: https://www.gstatic.com/firebasejs/9.18.0/firebase-auth.js 1:26682                                                B
https://www.gstatic.com/firebasejs/9.18.0/firebase-auth.js 1:26145                                                z
https://www.gstatic.com/firebasejs/9.18.0/firebase-auth.js 1:27195                                                Q
packages/firebase_auth_web/src/interop/auth.dart 40:42                                                            getAuthInstance
packages/firebase_auth_web/firebase_auth_web.dart 147:31                                                          get delegate
packages/firebase_auth_web/firebase_auth_web.dart 69:8                                                            new
packages/firebase_auth_web/firebase_auth_web.dart 156:12                                                          delegateFor
packages/firebase_auth_platform_interface/src/platform_interface/platform_interface_firebase_auth.dart 57:10      instanceFor
my_project/firebase_auth.dart 27:55                                                               get [_delegate]
my_project/firebase_auth.dart 518:25                                                              signInWithCredential
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54   runBody
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 123:5   _async
my_project/firebase_auth.dart 514:46                                                              signInWithCredential
my_project/firebase/auth_service.dart 83:59                                                         signInWithGoogle
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 45:50   <fn>
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/async/zone.dart 1660:54                                 runUnary
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/async/future_impl.dart 147:18                           handleValue
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/async/future_impl.dart 767:44                           handleValueCallback
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/async/future_impl.dart 796:13                           _propagateToListeners
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/async/future_impl.dart 567:5                            [_completeWithValue]
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/async/future.dart 528:22                                <fn>
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/async/zone.dart 1660:54                                 runUnary
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/async/future_impl.dart 147:18                           handleValue
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/async/future_impl.dart 767:44                           handleValueCallback
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/async/future_impl.dart 796:13                           _propagateToListeners
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/async/future_impl.dart 431:9                            callback
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/async/schedule_microtask.dart 40:11                     _microtaskLoop
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/async/schedule_microtask.dart 49:5                      _startMicrotaskLoop
C:/b/s/w/ir/cache/builder/src/out/host_debug/dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 166:15  <fn>

</details>

It was supposed to be fixed in the recent lib version but it's still not working. I'm using a Google Auth login. Same code works on mobile. Same code worked before (as mentioned in the closed issue)

I've now updated it to the most recent version with:
```
  firebase_core: 2.12.0
  firebase_auth: 4.6.0
  google_sign_in: 6.1.0
 
```
(FYI: firebase_auth_web = 5.5.0)

Thanks in advance for any help.

### PR DESCRIPTION
**Pull request by exaby73**
## Description

When the plugin is initialised by Flutter, initialise `FirebaseAuth` without any persistence. This causes a error to be thrown when calling `FirebaseAuth.instanceFor` with a persistence value as firebase_auth tries to reinitialise the auth instance with a new persistence. 

We cannot use `setPersistence` to fix this since it's a `Future` and `instanceFor` is synchronous, and the user cannot provide the persistence they want before the plugin is initialised. Therefore, the `persistence` parameter is being deprecated and users are expected to call `setPersistence` separately to set the `persistence` of the auth object.

Only affects web

## Related Issues

Fixes #10961

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_auth/firebase_auth_web/lib/src/interop/auth.dart [hunk 1]
```diff
@@ -57,7 +34,15 @@ Auth getAuthInstance(App app, {Persistence? persistence}) {
           auth_interop.browserSessionPersistence
         ],
         'popupRedirectResolver': auth_interop.browserPopupRedirectResolver
-      })));
+      }),
+    ),
+  );
+
+  if (persistence != null) {
+    auth.setPersistence(persistence);
```
Comment (russellwheatley) [inline packages/firebase_auth/firebase_auth_web/lib/src/interop/auth.dart:42]: this isn't going to work as we still need to use the `auth_interop.*` value for setting persistence. e.g. `auth_interop.indexedDBLocalPersistence`

**Review by russellwheatley — CHANGES_REQUESTED** [review]

File: packages/firebase_auth/firebase_auth_web/lib/src/interop/auth.dart (hunk 1 above)
Comment (russellwheatley) [reply packages/firebase_auth/firebase_auth_web/lib/src/interop/auth.dart:42]: do you recall that we were able to set `initializeAuth` twice without it erroring? We need to figure out if we can do that instead 🤔

Comment (exaby73) [reply packages/firebase_auth/firebase_auth_web/lib/src/interop/auth.dart:42]: I'll circle back to this soon. But as an answer to your first question, `setPersistence` is using `auth_interop.*` in it's implementation if I am not mistaken.

**Review by russellwheatley — APPROVED** [review]

File: packages/firebase_auth/firebase_auth/lib/src/firebase_auth.dart [hunk 2]
```diff
@@ -47,11 +41,15 @@ class FirebaseAuth extends FirebasePluginPlatform {
...
+  factory FirebaseAuth.instanceFor({
+    required FirebaseApp app,
+    @Deprecated(
+      'Will be removed in future release. Use setPersistence() instead.',
+    )
+    Persistence? persistence,
+  }) {
     return _firebaseAuthInstances.putIfAbsent(app.name, () {
-      return FirebaseAuth._(app: app, persistence: persistence);
+      return FirebaseAuth._(app: app);
```
Comment (Lyokone) [inline packages/firebase_auth/firebase_auth/lib/src/firebase_auth.dart:52]: If we remove the behavior, it's not a deprecation but a straight removal (breaking) even if we are warning the user.
we should just put the warning now, and remove the behavior at the next breaking release

**Pull request merged**